config/tests/src/test/java/org/jboss/windup/config/TestInsiderRuleProvider.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.windup.config;
+
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.graph.GraphContext;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ */
+public abstract class TestInsiderRuleProvider extends WindupRuleProvider implements Insider {
```

Comment:
This class should extend from both GraphCondition and GraphOperation, and should call both evaluate() and perform() methods like you've done below.

GraphOperation and GraphCondition are abstract classes, so you'll need to add an interface that defines the same methods that they each contain. 

Rename GraphCondition and GraphOperation to AbstractGraphCondition and AbstractGraphOperation
Create the interfaces named GraphCondition and GraphOperation.